npm/webpack-dev-server/src/devServer.ts

```diff
@@ -27,12 +27,16 @@ type FrameworkConfig = {
   }
 }
 
+export type ConfigHandler = Partial<Configuration>
+| ((baseConfig: Configuration) => Promise<Configuration>)
+| ((baseConfig: Configuration) => Configuration)
```

Comment:
Thoughts on consolidating the two function types and just vary the return type?

```
(baseConfig: Configuration) => (Configuration | Promise<Configuration>)
```

Side note: Should the return types be `Promise<Partial<Configuration>>` and `Partial<Configuration`?

Reply:
Oh nice, this is obviously better than what I had - I don't know why I wrote it like this :man_facepalming:

Reply:
I think we don't need `Partial` at all - every property in `Configuration` is nullable already (via webpack's type defs) but I'll add it for good measure, since we seem to do it in a few other places.